docs/src/FAQ.md

```diff
@@ -76,3 +76,52 @@ ChainRulesTestUtils.jl has some dependencies, so it is a separate package from C
 This means your package can depend on the light-weight ChainRulesCore.jl, and make ChainRulesTestUtils.jl a test-only dependency.
 
 Remember to read the section on [On writing good `rrule` / `frule` methods](@ref).
+
+## How do chain rules work for complex functions?
+
+`ChainRules.jl` follows the convention that `frule` applied to a function ``f(x + i y) = u(x,y) + i \, v(x,y)`` with perturbation ``\dot x + i \dot y`` returns the value and 
```

Comment:
I think that is too general of a statement. The way you describe it only really applies for `frule` if your input is real or `rrule` if your eventual output is real. I find it quite useful to think of forward-mode AD as propagating a perturbation in the input and getting out a derivative wrt to that input and reverse-mode as propagating a sensitivity in the output and getting out a vector of directional derivatives. Those perturbations/sensitivities can be complex, but not the input/output itself.

Reply:
The point of this FAQ is to point out two things:

1) `frule` and `rrule` are always defined (e.g. even for non-holomorphic `Complex -> Complex` functions). 
2) These definitions are sufficient to compute all quantities of interest. 

I've added a bit more details to the documentation, I hope this helps to clarify.

Reply:
I guess your comment arose because this convention for complex `frule` and `rrule` leads to a really awkward API for e.g. extracting Jacobians. This is definitely an issue, but the right place to fix this is probably in another package built on top of `ChainRules`, not in `ChainRules` itself.

Reply:
> These definitions are sufficient to compite all quantities of interest.

Depends on what you are interested in. Like I said above, only if the input/output is real will you get a meaningful result. For `C -> C` you would either want the Wirtinger primal and conjugate or a 2x2 Jacobian, which we don't currently support. We should clarify that, so that people don't get any wrong impressions.

Reply:
I have added a note at the very end which I hope addresses your concern. Let me know what you think.